### ISSUE TITLE
GeoRadius results

### ISSUE THREAD

**Issue by stevetayloruk** [StackExchange/StackExchange.Redis#724]
Hi, I'm trying to do a simple GeoRadius command but i get the following error:

Code: 
```
GeoRadiusResult[] result = db.GeoRadius("stockists", -1.759925, 52.19493, GeoUnit.Miles, 500, Order.Ascending, GeoRadiusOptions.WithDistance);
```

ERROR:
```
ERR could not decode requested zset member
```

Any assistance would be really grateful. I'm using Azure Redis Cache.

I can call this ok from the redis CLI.

thanks

Steve

**Comment by mgravell** [comment]
how was the data added?

**Comment by stevetayloruk** [comment]
```
 db.GeoAdd("stockists", -1.759925, 52.19493, "steve");
 db.GeoAdd("stockists", -3.360655, 54.66395, "dave");
```

**Comment by stevetayloruk** [comment]
Hi @mgravell,

Do you need more infomation from me about this issue? Is it something I'm doing wrong or is it a bug?

Thanks

Steve

**Comment by stevetayloruk** [comment]
UPDATE: It works if I pass in a GeoEntry Member into the GeoRadius method.

**Comment by NickCraver** [comment]
Can't readily test this on Windows at the moment (no Geo support), but we need to add a test here. Assigning to me - I'll spin it up when back at a desktop. Thanks for posting the workaround @stevetayloruk!

**Comment by mgravell** [comment]
me to confirm fixed before 2.0 ships (#871)

**Comment by NickCraver** [comment]
I see the problem here, implicit conversions were biting. This call:
```c#
db.GeoRadius(key, -1.759925, 52.19493, GeoUnit.Miles, 500, Order.Ascending, GeoRadiusOptions.WithDistance);
```
Matches this signature:
```c#
GeoRadius(RedisKey key, RedisValue member, double radius, GeoUnit unit, int count, Order? order, GeoRadiusOptions options, CommandFlags flags = CommandFlags.None);
```
This happens because `-1.759925` happily converts to `RedisValue` without issue. The overload *intended* here is:
```c#
GeoRadius(RedisKey key, double longitude, double latitude, double radius, GeoUnit unit, int count = -1, Order? order = null, GeoRadiusOptions options = GeoRadiusOptions.Default, CommandFlags flags = CommandFlags.None);
```
...which can be accomplished by:
```c#
GeoRadiusResult[] result = db.GeoRadius(key, -1.759925, 52.19493, 500, unit: GeoUnit.Miles, order: Order.Ascending, options: GeoRadiusOptions.WithDistance);
```
...or passing the `count` parameter. The reason this doesn't resolve to the signature already is the missing `count` parameter.

To be fair, this confused me on the first 2 glances. Perhaps we should break this overload so that the combination can't happen so easily. Or, we can say this was a one-time issue and wait to see if anyone hits it again. Thoughts?

**Comment by NickCraver** [comment]
@mgravell thought: if `RedisValue member` is double, throw? We've got the pattern elsewhere.

**Comment by mgravell** [comment]
Yeah, we can totally do that; I think it is *exceptionally* unlikely that anyone doing that is doing so intentionally

**Comment by NickCraver** [comment]
This is now fixed for 2.0 via dacf844582fae81bd21f7abe055dd1f98617da64 - closing out to cleanup. And thanks for the issue!
